go/vendor/github.com/keybase/go-updater/watchdog/watchdog.go

```diff
@@ -57,15 +60,81 @@ func Watch(programs []Program, restartDelay time.Duration, log Log) error {
 	return nil
 }
 
+func pidFileIsDeletable(terminatedPids []int, path string, log Log) (deletable bool) {
+	pidFromFile, err := ioutil.ReadFile(path)
+	if os.IsNotExist(err) {
+		// pidfile was successfully cleaned up by the terminated process
+		// or was never created
+		return false
+	}
+	lockedPid, err := strconv.Atoi(string(pidFromFile))
+	if err != nil {
+		log.Infof("error reading pid from file after terminating program %s: %s", path, err.Error())
+		return false
+	}
+	var terminatedPidIsStillLockedByFile bool
+	for _, termPid := range terminatedPids {
+		if termPid == lockedPid {
+			terminatedPidIsStillLockedByFile = true
+			break
+		}
+	}
+	return terminatedPidIsStillLockedByFile
+}
+
+func (p Program) deletePidFileIfTerminated(terminatedPids []int, log Log) {
+	if p.PidFile == "" {
+		// there is no pidfile on this program
+		return
+	}
+	var err error
+	timeout := time.NewTimer(3 * time.Second)
```

Comment:
how is it different from `time.After`? Just curious, I don't think I've seen it

Reply:
looks like `time.After` is a convenience around timers, and the advantage of using the underlying is that you have a little more control to do things like stop the timer when you know you don't care anymore (and allow it to be garbage collected). the way this is written, no difference at all.